buildpackage/builder.go

```diff
@@ -0,0 +1,138 @@
+package buildpackage
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/buildpack/imgutil"
+	"github.com/pkg/errors"
+
+	"github.com/buildpack/pack/dist"
+	"github.com/buildpack/pack/style"
+)
+
+type ImageFactory interface {
+	NewImage(repoName string, local bool) (imgutil.Image, error)
+}
+
+type PackageBuilder struct {
+	defaultBpInfo dist.BuildpackInfo
+	buildpacks    []dist.Buildpack
+	stacks        []dist.Stack
+	imageFactory  ImageFactory
+}
+
+func NewBuilder(imageFactory ImageFactory) *PackageBuilder {
```

Comment:
I hate how overloaded the term "builder" is in our code, but I suppose we can't get away from that at the moment...